EIPS/eip-2583.md

```diff
@@ -0,0 +1,295 @@
+DoS attacks, since they bypass cache mechanisms, thus creating a large discrepancy between 'normal' mode of execution and 'worst-case' execution of an opcode. 
+
+## Motivation
+
+As the ethereum trie becomes more and more saturated, the number of disk lookups that a node is required to do in order to access a piece of state increases too. This means that checking e.g. `EXTCODEHASH` of an account at block `5` was _inherently_ a cheaper operation that it is at, say `8.5M`. 
+
+From an implementation perspective, a node can (and does) use various caching mechanisms to cope with the problem, but there's an inherent problem with caches: when they yield a 'hit', they're great, but when they 'miss', they're useless. 
+
+This is attackable. By forcing a node to lookup non-existent keys, an attacker can maximize the number of disk lookups. 
+Sidenote: even if the 'non-existence' is cached, it's trivial to use a new non-existent key the next time, and never hit the same non-existent key again. Thus, caching 'non-existence' might be dangerous, since it will evict 'good' entries. 
+
+So far, the attempts to handle this problem has been in raising the gas cost, e.g. [EIP-150](https://eips.ethereum.org/EIPS/eip-150), [EIP-1884](https://eips.ethereum.org/EIPS/eip-1884). 
+
+However, when determining gas-costs, a secondary problem that arises due to the large discrepancy between 'happy-path' and 'notorious path' -- how do we determine the pricing?
+
+- The 'happy-path', assuming all items are cached? 
+  - Doing so would that would underprice all trie-accesses, and could be DoS-attacked. 
+- The 'normal' usage, based on benchmarks of actual usage? 
+  - This is basically what we do now, but that means that intentionally notorious executions are underpriced -- which constitutes a DoS vulnerability.
+- The 'paranoid' case: price everything as if caching did not exist?
+  - This would severely harm basically every contract due to the gas-cost increase. Also, if the gas limits were raised in order to allow the same amount of computation as before, the notorious case could again be used for DoS attacks.  
+
+From an engineering point of view, a node implementor is left with few options: 
+
+- Implement bloom filters for existence. This is difficult, not least because of the problems of reorgs, and the fact that it's difficult to undo bloom filter modifications. 
+- Implement flattened account databases. This is also difficult, both because of reorgs and also because it needs to be an _additional_ data structure aside from the `trie` -- we need the `trie` for consensus. So it's an extra data structure of around `15G` that needs to be kept in check. This is currently being pursued by the Geth-team.  
+
+This EIP proposes a mechanism to alleviate the situation. 
+
+## Specification
+
+We define the constant `penalty` as `TBD` (suggested `2000` gas).
```

Comment:
```suggestion
We define the constant `penalty` as `2000`.
```
Recommended during Draft (not required until Last Call): EIPs are assertions, put the number you want in and you can always change it later (while it is in draft).  Alternatively, if you don't want a sticky default just put in `TBD` and discuss `2000` in the discussion-to link.